docs/extensions/guides/kube-object-list-layout.md

```diff
@@ -1,3 +1,279 @@
+
+### Get CRD objects
+
+In order to list CRD objects, we need first fetch those from Kubernetes API. Lens Extensions API provides easy mechanism to do this. We just need to define `Certificate` class derived from `K8sApi.KubeObject`, `CertificatesApi`derived from `K8sApi.KubeApi` and `CertificatesStore` derived from `K8sApi.KubeObjectStore`.
+
+`Certificate` class defines properties found in the CRD object:
+
+```typescript
+export class Certificate extends K8sApi.KubeObject {
+  static kind = "Certificate"
+  static namespaced = true
+  static apiBase = "/apis/cert-manager.io/v1alpha2/certificates"
+
+  kind: string
+  apiVersion: string
+  metadata: {
+    name: string;
+    namespace: string;
+    selfLink: string;
+    uid: string;
+    resourceVersion: string;
+    creationTimestamp: string;
+    labels: {
+      [key: string]: string;
+    };
+    annotations: {
+      [key: string]: string;
+    };
+  }
+  spec: {
+    dnsNames: string[];
+    issuerRef: {
+      group: string;
+      kind: string;
+      name: string;
+    }
+    secretName: string
+  }
+  status: {
+    conditions: {
+      lastTransitionTime: string;
+      message: string;
+      reason: string;
+      status: string;
+      type?: string;
+    }[];
+  }
+}
+```
+
+With `CertificatesApi` class we are able to manage `Certificate` objects in Kubernetes API:
+
+```typescript
+export class CertifcatesApi extends K8sApi.KubeApi<Certificate> {
+}
+export const certificatesApi = new CertifcatesApi({
+  objectConstructor: Certificate
+});
+```
+
+`CertificateStore` defines storage for `Certificate` objects
+
+```typescript
+export class CertificatesStore extends K8sApi.KubeObjectStore<Certificate> {
+  api = certificatesApi
+}
+
+export const certificatesStore = new CertificatesStore();
+```
+
+And, finally, we register this store to Lens's API manager.
+
+```typescript
+K8sApi.apiManager.registerStore(certificatesStore);
+```
+
+
+### Create CertificatePage component
+
+Now we have created mechanism to manage `Certificate` objects in Kubernetes API. Then we need to fetch those and render them in the UI.
+
+First we define `CertificatePage` class that extends `React.Component`.
+
+```
+import { Component, LensRendererExtension } from "@k8slens/extensions";
+import React from "react";
+import { certificatesStore } from "../certificate-store";
+import { Certificate } from "../certificate"
+
+export class CertificatePage extends React.Component<{ extension: LensRendererExtension }> {
+
+}
+```
+
+We want to load all `Certificate` objects from `certificateStore` when the component is mounted. We can just call `certificatesStore.loadAll()` method:
+
+```
+export class CertificatePage extends React.Component<{ extension: LensRendererExtension }> {
+
+  async componentDidMount() {
+    await certificatesStore.loadAll()
+  }
+}
+```
```

Comment:
Probably we don't need this parts, since `<ItemListLayout />` component which is wrapperd by `KubeObjectListLayout />` is automatically loads all objects from its stores passed as a prop.
See https://github.com/lensapp/lens/blob/master/src/renderer/components/item-object-list/item-list-layout.tsx#L121

Reply:
Cool thanks! Did not know that.